File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/fragment/entitieslist/core/trashlist/impl/TrashListPresenterImpl.java
package com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.core.trashlist.impl;

import com.github.android.lvrn.lvrnproject.service.core.NoteService;
import com.github.android.lvrn.lvrnproject.service.form.NoteForm;
import com.github.android.lvrn.lvrnproject.util.CurrentState;
import com.github.android.lvrn.lvrnproject.util.PaginationArgs;
import com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.core.trashlist.TrashListPresenter;
import com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.impl.EntitiesListWithSearchPresenterImpl;
import com.github.valhallalabs.laverna.persistent.entity.Note;

import java.util.List;

/**
 * @author Andrii Bei <psihey1@gmail.com>
 */
public class TrashListPresenterImpl extends EntitiesListWithSearchPresenterImpl<Note, NoteForm> implements TrashListPresenter {

    private NoteService mNoteService;

    public TrashListPresenterImpl(NoteService entityService) {
        super(entityService);
        mNoteService = entityService;
    }

    @Override
    public void removeNoteForever() {
//        TODO: mNoteService.remove(NoteId);
    }

    @Override
    public void restoreNote() {
//        TODO: mNoteService.restoreFromTrash(noteId);
    }

    @Override
    protected List<Note> loadMoreForPagination(PaginationArgs paginationArgs) {
        return mNoteService.getTrashByProfile(CurrentState.Companion.getProfileId(), paginationArgs);
    }

    @Override
    protected List<Note> loadMoreForSearch(String query, PaginationArgs paginationArgs) {
        return mNoteService.getTrashByTitle(CurrentState.Companion.getProfileId(), query, paginationArgs);
    }
}

File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/fragment/entitieslist/core/trashlist/impl/TrashListPresenterImpl.kt
package com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.core.trashlist.impl

import com.github.android.lvrn.lvrnproject.service.core.NoteService
import com.github.android.lvrn.lvrnproject.service.form.NoteForm
import com.github.android.lvrn.lvrnproject.util.CurrentState.Companion.profileId
import com.github.android.lvrn.lvrnproject.util.PaginationArgs
import com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.core.trashlist.TrashListPresenter
import com.github.android.lvrn.lvrnproject.view.fragment.entitieslist.impl.EntitiesListWithSearchPresenterImpl
import com.github.valhallalabs.laverna.persistent.entity.Note
import javax.inject.Inject

class TrashListPresenterImpl @Inject constructor(var noteService: NoteService) : EntitiesListWithSearchPresenterImpl<Note, NoteForm>(noteService), TrashListPresenter {

    override fun loadMoreForPagination(paginationArgs: PaginationArgs): List<Note> {
        return noteService.getTrashByProfile(profileId!!, paginationArgs)
    }

    override fun loadMoreForSearch(query: String, paginationArgs: PaginationArgs): List<Note> {
        return noteService.getTrashByTitle(profileId!!, query, paginationArgs)
    }

    override fun removeNoteForever(position: Int) {
        noteService.removeForPermanent(mEntities.get(position).id)
        mEntities.removeAt(position)
        mEntitiesListFragment.updateRecyclerView()
    }

    override fun restoreNote(position: Int) {
        noteService.restoreFromTrash(mEntities.get(position).id)
        mEntities.removeAt(position)
        mEntitiesListFragment.updateRecyclerView()
    }


}